### PR TITLE
[#IC-489] migrate ts-lint to eslint for io-functions-public

### PR DESCRIPTION
**Pull request by arcogabbo**
#### List of Changes

Migrated project from tslint to eslint

#### Motivation and Context

Development purposes

#### How Has This Been Tested?

All tests run fine

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: utils/healthcheck.ts [hunk 1]
```diff
@@ -135,7 +137,8 @@ export const checkApplicationHealth = (): HealthCheck<ProblemSource, true> =>
       // TODO: once we upgrade to fp-ts >= 1.19 we can use Validation to collect all errors, not just the first to happen
       sequenceT(taskEither)<
         ReadonlyArray<HealthProblem<ProblemSource>>,
-        // tslint:disable readonly-array beacuse the following is actually mutable
+        // eslint-disable functional/prefer-readonly-type
+        // eslint-disable-next-line functional/prefer-readonly-type
```
Comment (gquadrati) [inline utils/healthcheck.ts:141]: duplicated
```suggestion
        // eslint-disable-next-line functional/prefer-readonly-type
```

**Review by gquadrati — CHANGES_REQUESTED** [review]
Some small comments

File: ValidateProfileEmail/handler.ts [hunk 2]
```diff
@@ -80,16 +82,19 @@ const TokenQueryParamMiddleware = RequiredQueryParamMiddleware(
   TokenQueryParam
 );
 
-// tslint:disable-next-line: cognitive-complexity
+// eslint-disable-next-line sonarjs/cognitive-complexity
```
[the remark this answers is not in the document]
Comment (michaeldisaro) [reply ValidateProfileEmail/handler.ts:79]: @arcogabbo just try to remove it and check if linting works.

**Review by michaeldisaro — APPROVED** [review]
LGTM. Just try to check the comment leave by @gquadrati .

**Pull request merged**